File: architecture/testing/across-languages/python/pricing.py
"""Доменное ядро — тот же код, что в go/pricing и java Pricing.

Объёмная скидка по тирам, всё в копейках.
"""

TIER1_CENTS = 10_000  # от 100.00 — 5%
TIER2_CENTS = 20_000  # от 200.00 — 10%


def discount(total_cents: int) -> int:
    """Размер скидки в копейках для суммы заказа."""
    if total_cents >= TIER2_CENTS:
        return total_cents * 10 // 100
    if total_cents >= TIER1_CENTS:
        return total_cents * 5 // 100
    return 0


def price(total_cents: int) -> int:
    """Итоговая цена: сумма минус скидка."""
    return total_cents - discount(total_cents)

File: architecture/testing/across-languages/go/pricing/pricing.go
// Package pricing — доменное ядро, одинаковое во всех трёх языках стенда:
// объёмная скидка по тирам и итоговая цена. Ровно этот код (построчно тот же
// смысл) есть в java/ и python/ — чтобы сравнивать не задачи, а экосистемы.
package pricing

// Тиры скидки: от 100.00 — 5%, от 200.00 — 10%. Всё в копейках.
const (
	Tier1Cents = 10_000
	Tier2Cents = 20_000
)

// Discount — размер скидки в копейках для суммы заказа.
func Discount(totalCents int64) int64 {
	switch {
	case totalCents >= Tier2Cents:
		return totalCents * 10 / 100
	case totalCents >= Tier1Cents:
		return totalCents * 5 / 100
	default:
		return 0
	}
}

// Price — итоговая цена: сумма минус скидка.
func Price(totalCents int64) int64 {
	return totalCents - Discount(totalCents)
}
